Installation issue: util-linux

<!-- Thanks for taking the time to report this build failure. To proceed with the report please:

1. Title the issue "Installation issue: <name-of-the-package>".
2. Provide the information required below.

We encourage you to try, as much as possible, to reduce your problem to the minimal example that still reproduces the issue. That would help us a lot in fixing it quickly and effectively! -->

### Steps to reproduce the issue

<!-- Fill in the exact spec you are trying to build and the relevant part of the error message -->
```console
$ spack install util-linux
==> pkg-config@0.29.1 : externally installed in /usr
[+] /usr
==> python@3.6.9 : externally installed in /usr
[+] /usr
==> Installing util-linux
==> No binary for util-linux found: installing from source
==> Using cached archive: /home/rblake/spack/var/spack/cache/_source-cache/archive/37/37ac05d82c6410d89bc05d43cee101fefc8fe6cf6090b3ce7a1409a6f35db606.tar.gz
==> util-linux: Executing phase: 'autoreconf'
==> util-linux: Executing phase: 'configure'
==> util-linux: Executing phase: 'build'
==> util-linux: Executing phase: 'install'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j8' 'install'

44 errors found in build log:
     1094     /bin/mkdir -p '/home/rblake/spack/opt/spack/linux-ubuntu18.04-ivybridge/gcc-8.4.0/util-linux-2.35.1-rdfgmclmwmzbmymlmjfv2tvwhmsjqdex/lib/p
             ython3.6/site-packages/libmount'
     1095     /usr/bin/install -c -m 644 libmount/python/__init__.py '/home/rblake/spack/opt/spack/linux-ubuntu18.04-ivybridge/gcc-8.4.0/util-linux-2.35
             .1-rdfgmclmwmzbmymlmjfv2tvwhmsjqdex/lib/python3.6/site-packages/libmount'
     1096     /bin/mkdir -p '/usr/share/bash-completion/completions'
     1097     /bin/mkdir -p '/home/rblake/spack/opt/spack/linux-ubuntu18.04-ivybridge/gcc-8.4.0/util-linux-2.35.1-rdfgmclmwmzbmymlmjfv2tvwhmsjqdex/share
             /doc/util-linux/getopt/'
     1098     /usr/bin/install -c -m 644 bash-completion/blkid bash-completion/cal bash-completion/col bash-completion/colcrt bash-completion/colrm bash
             -completion/column bash-completion/fincore bash-completion/findfs bash-completion/findmnt bash-completion/flock bash-completion/getopt bash
             -completion/hexdump bash-completion/ipcmk bash-completion/ipcrm bash-completion/ipcs bash-completion/isosize bash-completion/logger bash-co
             mpletion/look bash-completion/lsipc bash-completion/lsns bash-completion/mcookie bash-completion/mkfs bash-completion/mkswap bash-completio
             n/namei bash-completion/prlimit bash-completion/readprofile bash-completion/renice bash-completion/rev bash-completion/script bash-completi
             on/scriptreplay bash-completion/scriptlive bash-completion/setsid bash-completion/whereis bash-completion/wipefs bash-completion/fsck bash-
             completion/mkfs.bfs bash-completion/fsck.minix bash-completion/mkfs.minix bash-completion/fsck.cramfs bash-completion/mkfs.cramfs '/usr/sha
             re/bash-completion/completions'
     1099     /usr/bin/install -c misc-utils/getopt-parse.bash misc-utils/getopt-parse.tcsh '/home/rblake/spack/opt/spack/linux-ubuntu18.04-ivybridge/gc
             c-8.4.0/util-linux-2.35.1-rdfgmclmwmzbmymlmjfv2tvwhmsjqdex/share/doc/util-linux/getopt/'
  >> 1100    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/blkid': Permission denied
  >> 1101    /usr/bin/install: cannot create regular file '/usr/share/bash-completion/completions/cal': Permission denied
  >> 1102    /usr/bin/install: cannot create regular file '/usr/share/bash-completion/completions/col': Permission denied
  >> 1103    /usr/bin/install: cannot create regular file '/usr/share/bash-completion/completions/colcrt': Permission denied
  >> 1104    /usr/bin/install: cannot create regular file '/usr/share/bash-completion/completions/colrm': Permission denied
  >> 1105    /usr/bin/install: cannot create regular file '/usr/share/bash-completion/completions/column': Permission denied
  >> 1106    /usr/bin/install: cannot create regular file '/usr/share/bash-completion/completions/fincore': Permission denied
  >> 1107    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/findfs': Permission denied
  >> 1108    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/findmnt': Permission denied
  >> 1109    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/flock': Permission denied
  >> 1110    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/getopt': Permission denied
  >> 1111    /usr/bin/install: cannot create regular file '/usr/share/bash-completion/completions/hexdump': Permission denied
  >> 1112    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/ipcmk': Permission denied
  >> 1113    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/ipcrm': Permission denied
  >> 1114    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/ipcs': Permission denied
  >> 1115    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/isosize': Permission denied
  >> 1116    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/logger': Permission denied
  >> 1117    /usr/bin/install: cannot create regular file '/usr/share/bash-completion/completions/look': Permission denied
  >> 1118    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/lsipc': Permission denied
  >> 1119    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/lsns': Permission denied
  >> 1120    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/mcookie': Permission denied
  >> 1121    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/mkfs': Permission denied
  >> 1122    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/mkswap': Permission denied
  >> 1123    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/namei': Permission denied
  >> 1124    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/prlimit': Permission denied
  >> 1125    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/readprofile': Permission denied
  >> 1126    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/renice': Permission denied
  >> 1127    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/rev': Permission denied
  >> 1128    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/script': Permission denied
  >> 1129    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/scriptreplay': Permission denied
  >> 1130    /usr/bin/install: cannot create regular file '/usr/share/bash-completion/completions/scriptlive': Permission denied
  >> 1131    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/setsid': Permission denied
  >> 1132    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/whereis': Permission denied
  >> 1133    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/wipefs': Permission denied
  >> 1134    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/fsck': Permission denied
  >> 1135    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/mkfs.bfs': Permission denied
  >> 1136    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/fsck.minix': Permission denied
  >> 1137    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/mkfs.minix': Permission denied
  >> 1138    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/fsck.cramfs': Permission denied
  >> 1139    /usr/bin/install: cannot remove '/usr/share/bash-completion/completions/mkfs.cramfs': Permission denied
     1140    Makefile:13181: recipe for target 'install-dist_bashcompletionDATA' failed
  >> 1141    make[3]: *** [install-dist_bashcompletionDATA] Error 1
     1142    make[3]: *** Waiting for unfinished jobs....
     1143     /bin/mkdir -p '/home/rblake/spack/opt/spack/linux-ubuntu18.04-ivybridge/gcc-8.4.0/util-linux-2.35.1-rdfgmclmwmzbmymlmjfv2tvwhmsjqdex/share
             /man/man1'
     1144     /bin/mkdir -p '/home/rblake/spack/opt/spack/linux-ubuntu18.04-ivybridge/gcc-8.4.0/util-linux-2.35.1-rdfgmclmwmzbmymlmjfv2tvwhmsjqdex/share
             /man/man3'
     1145     /usr/bin/install -c -m 644 libuuid/man/uuid.3 libuuid/man/uuid_clear.3 libuuid/man/uuid_compare.3 libuuid/man/uuid_copy.3 libuuid/man/uuid
             _generate.3 libuuid/man/uuid_is_null.3 libuuid/man/uuid_parse.3 libuuid/man/uuid_time.3 libuuid/man/uuid_unparse.3 libuuid/man/uuid_generat
             e_random.3 libuuid/man/uuid_generate_time.3 libuuid/man/uuid_generate_time_safe.3 libblkid/libblkid.3 '/home/rblake/spack/opt/spack/linux-u
             buntu18.04-ivybridge/gcc-8.4.0/util-linux-2.35.1-rdfgmclmwmzbmymlmjfv2tvwhmsjqdex/share/man/man3'
     1146     /usr/bin/install -c -m 644 schedutils/chrt.1 schedutils/ionice.1 schedutils/taskset.1 text-utils/col.1 text-utils/colcrt.1 text-utils/colr
             m.1 text-utils/column.1 text-utils/hexdump.1 text-utils/rev.1 text-utils/ul.1 text-utils/more.1 term-utils/script.1 term-utils/scriptreplay
             .1 term-utils/scriptlive.1 term-utils/setterm.1 term-utils/mesg.1 term-utils/wall.1 login-utils/last.1 login-utils/lastb.1 login-utils/utmp
             dump.1 login-utils/lslogins.1 sys-utils/lsmem.1 sys-utils/flock.1 sys-utils/choom.1 sys-utils/ipcmk.1 sys-utils/ipcrm.1 sys-utils/ipcs.1 sy
             s-utils/lsipc.1 sys-utils/renice.1 sys-utils/setsid.1 sys-utils/dmesg.1 sys-utils/eject.1 sys-utils/prlimit.1 sys-utils/lscpu.1 sys-utils/m
             ountpoint.1 sys-utils/fallocate.1 sys-utils/unshare.1 sys-utils/nsenter.1 misc-utils/cal.1 misc-utils/logger.1 '/home/rblake/spack/opt/spac
             k/linux-ubuntu18.04-ivybridge/gcc-8.4.0/util-linux-2.35.1-rdfgmclmwmzbmymlmjfv2tvwhmsjqdex/share/man/man1'
     1147     /usr/bin/install -c -m 644 misc-utils/look.1 misc-utils/mcookie.1 misc-utils/namei.1 misc-utils/whereis.1 misc-utils/uuidgen.1 misc-utils/
             uuidparse.1 misc-utils/kill.1 misc-utils/rename.1 misc-utils/getopt.1 misc-utils/fincore.1 misc-utils/hardlink.1 '/home/rblake/spack/opt/sp
             ack/linux-ubuntu18.04-ivybridge/gcc-8.4.0/util-linux-2.35.1-rdfgmclmwmzbmymlmjfv2tvwhmsjqdex/share/man/man1'
     1148    Byte-compiling python modules...
     1149    __init__.py
     1150    Byte-compiling python modules (optimized versions) ...
     1151    __init__.py
     1152    make[3]: Leaving directory '/tmp/rblake/spack-stage/spack-stage-util-linux-2.35.1-rdfgmclmwmzbmymlmjfv2tvwhmsjqdex/spack-src'
     1153    Makefile:13668: recipe for target 'install-am' failed
  >> 1154    make[2]: *** [install-am] Error 2
     1155    make[2]: Leaving directory '/tmp/rblake/spack-stage/spack-stage-util-linux-2.35.1-rdfgmclmwmzbmymlmjfv2tvwhmsjqdex/spack-src'
     1156    Makefile:13356: recipe for target 'install-recursive' failed
  >> 1157    make[1]: *** [install-recursive] Error 1
     1158    make[1]: Leaving directory '/tmp/rblake/spack-stage/spack-stage-util-linux-2.35.1-rdfgmclmwmzbmymlmjfv2tvwhmsjqdex/spack-src'
     1159    Makefile:13662: recipe for target 'install' failed
  >> 1160    make: *** [install] Error 2

See build log for details:
  /tmp/rblake/spack-stage/spack-stage-util-linux-2.35.1-rdfgmclmwmzbmymlmjfv2tvwhmsjqdex/spack-build-out.txt


```

### Information on your system

<!-- Please include the output of `spack debug report` -->
* **Spack:** 0.15.4-1005-0d5c06567
* **Python:** 3.6.9
* **Platform:** linux-ubuntu18.04-ivybridge

<!-- If you have any relevant configuration detail (custom `packages.yaml` or `modules.yaml`, etc.) you can add that here as well. -->

### Additional information

<!-- Please upload the following files. They should be present in the stage directory of the failing build. Also upload any config.log or similar file if one exists. -->
* [spack-build-out.txt](https://github.com/spack/spack/files/5233443/spack-build-out.txt)
* [spack-build-env.txt](https://github.com/spack/spack/files/5233446/spack-build-env.txt)


<!-- Some packages have maintainers who have volunteered to debug build failures. Run `spack maintainers <name-of-the-package>` and @mention them here if they exist. -->

### General information

<!-- These boxes can be checked by replacing [ ] with [x] or by clicking them after submitting the issue. -->
- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/cryptsetup/package.py
+++ b/var/spack/repos/builtin/packages/cryptsetup/package.py
@@ -25,7 +25,7 @@ class Cryptsetup(AutotoolsPackage):
     depends_on('lvm2', type=('build', 'link'))
     depends_on('popt', type=('build', 'link'))
     depends_on('json-c', type=('build', 'link'))
-    depends_on('util-linux', type=('build', 'link'))
+    depends_on('util-linux~libuuid', type=('build', 'link'))
     depends_on('gettext', type=('build', 'link'))
 
     depends_on('pkgconfig', type='build')

--- a/var/spack/repos/builtin/packages/cryptsetup/package.py
+++ b/var/spack/repos/builtin/packages/cryptsetup/package.py
@@ -25,7 +25,7 @@ class Cryptsetup(AutotoolsPackage):
     depends_on('lvm2', type=('build', 'link'))
     depends_on('popt', type=('build', 'link'))
     depends_on('json-c', type=('build', 'link'))
-    depends_on('util-linux~libuuid', type=('build', 'link'))
+    depends_on('util-linux', type=('build', 'link'))
     depends_on('gettext', type=('build', 'link'))
 
     depends_on('pkgconfig', type='build')
@@ -35,6 +35,7 @@ class Cryptsetup(AutotoolsPackage):
     depends_on('m4',       type='build')
 
     depends_on('automake@:1.16.1', when='@2.2.1', type='build')
+    depends_on('openssl')
 
     # Upstream includes support for discovering the location of the libintl
     # library but is missing the bit in the Makefile.ac that includes it in
@@ -49,7 +50,8 @@ class Cryptsetup(AutotoolsPackage):
 
     def configure_args(self):
         args = [
-            'systemd_tmpfilesdir={0}/tmpfiles.d'.format(self.prefix)
+            'systemd_tmpfilesdir={0}/tmpfiles.d'.format(self.prefix),
+            '--with-crypto_backend=openssl',
         ]
         return args
 

--- a/var/spack/repos/builtin/packages/util-linux/package.py
+++ b/var/spack/repos/builtin/packages/util-linux/package.py
@@ -43,8 +43,9 @@ class UtilLinux(AutotoolsPackage):
             '--without-systemd',
         ]
         if "+bash" in self.spec:
-            config_args.extend(['--enable-bash-completion',
-                                '--bash-completion-dir=',+self.spec['bash'].prefix])
+            config_args.extend(
+                ['--enable-bash-completion',
+                 '--bash-completion-dir=' + self.spec['bash'].prefix])
         else:
             config_args.append('--disable-bash-completion')
         config_args.extend(self.enable_or_disable('libuuid'))

--- a/var/spack/repos/builtin/packages/util-linux/package.py
+++ b/var/spack/repos/builtin/packages/util-linux/package.py
@@ -48,8 +48,8 @@ class UtilLinux(AutotoolsPackage):
             config_args.extend(
                 ['--enable-bash-completion',
                  '--with-bashcompletiondir=' + os.path.join(
-                     self.spec['bash'].prefix,"share","bash-completion","completions"
-                     )])
+                     self.spec['bash'].prefix,
+                     "share", "bash-completion", "completions")])
         else:
             config_args.append('--disable-bash-completion')
         config_args.extend(self.enable_or_disable('libuuid'))

--- a/var/spack/repos/builtin/packages/util-linux/package.py
+++ b/var/spack/repos/builtin/packages/util-linux/package.py
@@ -31,7 +31,7 @@ class UtilLinux(AutotoolsPackage):
     variant('libuuid', default=True, description='Build libuuid')
     variant('bash', default=False, description='Install bash completion scripts')
 
-    depends_on('bash', when="+bash")
+    depends_on('bash', when="+bash", type='run')
     depends_on('libuuid', when="+libuuid")
 
     def url_for_version(self, version):

--- a/var/spack/repos/builtin/packages/util-linux/package.py
+++ b/var/spack/repos/builtin/packages/util-linux/package.py
@@ -32,7 +32,6 @@ class UtilLinux(AutotoolsPackage):
     variant('bash', default=False, description='Install bash completion scripts')
 
     depends_on('bash', when="+bash", type='run')
-    depends_on('libuuid', when="+libuuid")
 
     def url_for_version(self, version):
         url = "https://www.kernel.org/pub/linux/utils/util-linux/v{0}/util-linux-{1}.tar.gz"

--- a/var/spack/repos/builtin/packages/util-linux/package.py
+++ b/var/spack/repos/builtin/packages/util-linux/package.py
@@ -4,6 +4,7 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 from spack import *
+import os
 
 
 class UtilLinux(AutotoolsPackage):
@@ -46,7 +47,9 @@ class UtilLinux(AutotoolsPackage):
         if "+bash" in self.spec:
             config_args.extend(
                 ['--enable-bash-completion',
-                 '--bash-completion-dir=' + self.spec['bash'].prefix])
+                 '--with-bashcompletiondir=' + os.path.join(
+                     self.spec['bash'].prefix,"share","bash-completion","completions"
+                     )])
         else:
             config_args.append('--disable-bash-completion')
         config_args.extend(self.enable_or_disable('libuuid'))

--- a/var/spack/repos/builtin/packages/util-linux/package.py
+++ b/var/spack/repos/builtin/packages/util-linux/package.py
@@ -31,6 +31,7 @@ class UtilLinux(AutotoolsPackage):
     variant('bash', default=False, description='Install bash completion scripts')
 
     depends_on('bash', when="+bash")
+    depends_on('libuuid', when="+libuuid")
 
     def url_for_version(self, version):
         url = "https://www.kernel.org/pub/linux/utils/util-linux/v{0}/util-linux-{1}.tar.gz"

--- a/var/spack/repos/builtin/packages/util-linux/package.py
+++ b/var/spack/repos/builtin/packages/util-linux/package.py
@@ -28,6 +28,9 @@ class UtilLinux(AutotoolsPackage):
     # Make it possible to disable util-linux's libuuid so that you may
     # reliably depend_on(`libuuid`).
     variant('libuuid', default=True, description='Build libuuid')
+    variant('bash', default=False, description='Install bash completion scripts')
+
+    depends_on('bash', when="+bash")
 
     def url_for_version(self, version):
         url = "https://www.kernel.org/pub/linux/utils/util-linux/v{0}/util-linux-{1}.tar.gz"
@@ -37,7 +40,13 @@ class UtilLinux(AutotoolsPackage):
         config_args = [
             '--disable-use-tty-group',
             '--disable-makeinstall-chown',
-            '--without-systemd'
+            '--without-systemd',
         ]
+        if "+bash" in self.spec:
+            config_args.extend(['--enable-bash-completion',
+                                '--bash-completion-dir=',+self.spec['bash'].prefix])
+        else:
+            config_args.append('--disable-bash-completion')
         config_args.extend(self.enable_or_disable('libuuid'))
+
         return config_args

